planning/behavior_velocity_planner/autoware_behavior_velocity_traffic_light_module/src/scene.cpp

```diff
@@ -194,22 +196,47 @@ bool TrafficLightModule::isStopSignal()
   }
 
   // Check if the current traffic signal state requires stopping
-  return autoware::traffic_light_utils::isTrafficSignalStop(lane_, looking_tl_state_);
+  return autoware::traffic_light_utils::isTrafficSignalStop(lane_, looking_tl_state_.elements);
+}
+
+bool TrafficLightModule::willTrafficLightTurnRedBeforeReachingStopLine(
+  const double & distance_to_stop_line) const
+{
+  double ego_velocity = planner_data_->current_velocity->twist.linear.x;
+  double predicted_passing_stop_line_time = ego_velocity > planner_param_.v2i_velocity_threshold
+                                              ? distance_to_stop_line / ego_velocity
+                                              : planner_param_.v2i_required_time_to_departure;
+
+  auto seconds = predicted_passing_stop_line_time - planner_param_.v2i_last_time_allowed_to_pass;
+
+  auto now = clock_->now();
+  // find stop signal from looking_tl_state_.predictions by using isTrafficSignalStop
+  for (const auto & prediction : looking_tl_state_.predictions) {
+    if (autoware::traffic_light_utils::isTrafficSignalStop(
+          lane_, prediction.simultaneous_elements)) {
+      if ((now - prediction.predicted_stamp).seconds() < seconds) {
```

Comment:
In my understanding, `prediction.predicted_stamp` has future timestamp. So this line should be like this?
```
if ((prediction.predicted_stamp - now).seconds() < seconds) {
```